include/api/CAnomalyJobConfig.h

```diff
@@ -249,19 +261,6 @@ class API_EXPORT CAnomalyJobConfig {
         //! Parse a JSON value representing an entire analysis config object.
         void parse(const json::value& json);
 
-        //! Return a JSON string representing the analysis config
-        const std::string& getAnalysisConfig();
-
-        //! Reparse the detector configuration object from within a stored
-        //! string representing the analysis config object.
-        //! This is necessary to correctly reinitialise scoped rule objects
-        //! folowing an update of the fiter rules configuration.
-        bool reparseDetectorsFromStoredConfig(const std::string& analysisConfig);
-
-        void setConfig(const std::string& analysisConfigString) {
-            m_AnalysisConfigString = analysisConfigString;
-        }
-
```

Comment:
Can you please leave a comment in this PR on why do you remove those?

Reply:
The code to `set`, `get` and `reparse` analysis config was introduced as part of the work to upgrade from boost 1.77 to 1.83. That upgrade included a major rewrite of boost unordered containers. Unfortunately it broke behaviour that we had come to rely on related to updating analysis config during a jobs' active lifetime. As a workaround the analysis config was reset to its previous state after an update, while allowing filter rules config to be updated. This PR provides a correct fix for the filter rules update, which also allows the analysis config to be updated safely as well. Therefore any code related to resetting the analysis config to its previous state is no longer needed.